src/Features/Core/Portable/SemanticSearch/AbstractSemanticSearchService.cs

```diff
@@ -262,17 +291,30 @@ private static (MethodInfo? method, QueryKind queryKind) GetFindMethod(Type type
                 return default;
             }
 
-            if (!s_queryKindByParameterType.TryGetValue(parameter.ParameterType, out var entity))
+            if (!s_queryKindByParameterType.TryGetValue(parameter.ParameterType, out var queryKind))
             {
                 error = string.Format(
-                    FeaturesResources.Type_0_is_not_among_supported_types_1,
-                    SemanticSearchUtilities.FindMethodName,
+                    FeaturesResources.Parameter_type_0_is_not_among_supported_types_1,
+                    parameter.ParameterType,
                     string.Join(", ", s_queryKindByParameterType.Keys.Select(t => $"'{t.Name}'")));
 
                 return default;
             }
 
-            return (method, entity);
+            // IEnumerable<ISymbol>
+            // IAsyncEnumerable<ISymbol>
+            bool? isAsync = method.ReturnType == typeof(IEnumerable<ISymbol>) ? false : method.ReturnType == typeof(IAsyncEnumerable<ISymbol>) ? true : null;
+            if (isAsync == null)
+            {
+                error = string.Format(
+                    FeaturesResources.Return_type_0_is_not_among_supported_types_1,
+                    method.ReturnType,
+                    $"'{typeof(IEnumerable<ISymbol>).Name}', '{typeof(IAsyncEnumerable<ISymbol>).Name}'");
```

Comment:
does .Name do a good job here?  i thought with instantiated generics it was something *awful* :)   do you have a test showing what error message we actually get? 

personally, it might be better to literally hardcode `"IEnumerable<ISymbol>"`.

--

Oh, looking at this in a console app, you just get ``IEnumerable`1``.  That doesn't seem great here.  Consider just hard coding the name.